proxy/gateway/nodejs/lib/channel.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+let EventEmitter = require('events').EventEmitter;
+let nano = require('nanomsg');
+let fs = require('fs');
+
+class Channel extends EventEmitter {
+  constructor() {
+    super();
+    this.socket = nano.socket('pair');
+    this.socket.on('data', data => this.emit('data', data));
+  }
+
+  connect(id) {
+    let addr = (typeof(id) === 'string') ? 'ipc://' + id : id.uri;
```

Comment:
Added a commit to throw when the message channel URI doesn't start with `ipc://`